admin/class-yoast-network-settings-api.php

```diff
@@ -0,0 +1,195 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Admin\Network
+ */
+
+/**
+ * Implements a network settings API for the plugin's multisite settings.
+ */
+class Yoast_Network_Settings_API {
+
+	/** Action identifier for updating plugin network options. */
+	const UPDATE_ACTION = 'yoast_handle_network_options';
+
+	/** @var array Registered network settings. */
+	private $registered_settings = array();
+
+	/** @var array Options whitelist, keyed by option group. */
+	private $whitelist_options = array();
+
+	/** @var bool Internal flag for whether the necessary hooks have been added. */
+	private $hooks_registered = false;
+
+	/** @var Yoast_Network_Settings_API The singleton instance of this class. */
+	private static $instance = null;
+
+	/**
+	 * Outputs nonce, action and option group fields for a network settings page in the plugin.
+	 *
+	 * @param string $option_group Option group name for the current page.
+	 */
+	public function settings_fields( $option_group ) {
+		?>
+		<input type="hidden" name="network_option_group" value="<?php echo esc_attr( $option_group ); ?>" />
+		<input type="hidden" name="action" value="<?php echo esc_attr( self::UPDATE_ACTION ); ?>" />
+		<?php
+		wp_nonce_field( "$option_group-network-options" );
+	}
+
+	/**
+	 * Registers a network setting and its data.
+	 *
+	 * @param string $option_group The group the network option is part of.
+	 * @param string $option_name  The name of the network option to sanitize and save.
+	 * @param array  $args         {
+	 *     Optional. Data used to describe the network setting when registered.
+	 *
+	 *     @type callable $sanitize_callback A callback function that sanitizes the network option's value.
+	 *     @type mixed    $default           Default value when calling `get_network_option()`.
+	 * }
+	 */
+	public function register_setting( $option_group, $option_name, $args = array() ) {
+
+		$args = wp_parse_args( $args, array(
+			'group'             => $option_group,
+			'sanitize_callback' => null,
+		) );
+
+		if ( ! isset( $this->whitelist_options[ $option_group ] ) ) {
+			$this->whitelist_options[ $option_group ] = array();
+		}
+
+		$this->whitelist_options[ $option_group ][] = $option_name;
+
+		if ( ! empty( $args['sanitize_callback'] ) ) {
+			add_filter( "sanitize_option_{$option_name}", array( $this, 'filter_sanitize_option' ), 10, 2 );
+		}
+		if ( array_key_exists( 'default', $args ) ) {
+			add_filter( "default_site_option_{$option_name}", array( $this, 'filter_default_option' ), 10, 2 );
+		}
+
+		$this->registered_settings[ $option_name ] = $args;
+	}
+
+	/**
+	 * Handles a request in which plugin network options should be updated.
```

Comment:
This suggests a certain level of uncertainty in it's functionality. Please refrain from using words like 'should'.

Reply:
I generally get what you're saying, but I'm not sure how to otherwise express the functionality of this method here. Using "will" would be inaccurate because that depends on whether the form data has been submitted correctly and whether settings are registered correctly.